### ISSUE TITLE
Incorrectly reporting errors

### ISSUE THREAD

**Issue by John-Whitehead** [rokucommunity/vscode-brightscript-language#226]
Operating System: Fedora 30
VS Code: 1.44.0
BRS Extension:  2.1.2

I am getting red squiggle lines under code which is actually correct.  It was on a huge non-SG screensaver `main.brs`, but I can reproduce it at will with the following steps.
`$ mkdir test`
`$ cd test`
`$ code .`

Then type or paste the following code:
```
Function RunScreenSaver( params As Object ) As Object
     main()
End Function

sub main()
    print "Entering main()..."
    m.screen  = CreateObject("roScreen")                    ' Required object
    m.port    = CreateObject("roMessagePort")               ' Required object
    di        = CreateObject("roDeviceInfo")                ' For resolution info
    timeStamp = CreateObject("roDateTime")                  ' For elapsed time
    i% = 1

    i% = i% << 3
    i% = i% >> 3

    for j = 1 to 10
        print j
    next j

    for k = 1 to 10
        print k
    end for

end sub
```
All should look well, until you hit `<CTRL-S>` and save as `main.brs.`  After writing to disk, the red marks appear, as shown in the attached screenshot below.

There was a red squiggle under the "j" in "next j" but I upgraded the version of VS Code and the extension this morning and I can no longer reproduce that.

My screensaver compiles and runs fine even with the squigle marks.

![Screenshot from 2020-04-12 09-56-33](https://user-images.githubusercontent.com/40612016/79070979-00060a80-7ca7-11ea-973d-6841eac7dd12.png)

**Comment by TwitchBronBron** [comment]
I'm a little confused about "Then type or paste the following code:". Do you mean open a new untitled file in vscode?
![image](https://user-images.githubusercontent.com/2544493/79074946-6f86f480-7cbd-11ea-91dc-e13f7e31dde9.png)

 If so, then the language validation doesn't actually work for those types of files. The file needs to have been saved with a .brs extension. So this all actually makes sense to me.

**Comment by John-Whitehead** [comment]
You are correct, Bronley.  You need to open a new file before pasting the code.  I updated my original comment. Yes, the errors do not appear until after the file is saved with a .brs extension and the status line at the bottom changes from "Plain Text" to "Brightscript".  So, the thought that it happens sometimes and not others was just a red herring.

**Comment by chrisdp** [comment]
Update to this issue: 
![image](https://user-images.githubusercontent.com/9591618/160266886-1e2a68a5-dd6f-4b5f-b847-c8c42dd5f29a.png)
![image](https://user-images.githubusercontent.com/9591618/160266898-2627f289-0521-45b0-982d-ac53cebcfadc.png)

**Comment by TwitchBronBron** [comment]
This is a weird syntax. What's the use case for that? Could you call `next <some other variable>`? If so, what does it do? In order to support this functionality, I need to know how it functions so the language server can build the AST.
